extensions/amp-next-page/0.2/page.js

```diff
@@ -112,14 +121,17 @@ export class Page {
   /**
    * @param {VisibilityState} visibilityState
    */
-  setVisible(visibilityState) {
+  setVisibility(visibilityState) {
     // TODO(wassgha): Handle history manipulation
     // TODO(wassgha): Handle manual visibility management
 
     // Update visibility internally and at the shadow doc level
-    if (this.shadowDoc_ && visibilityState != this.visibilityState_) {
-      this.shadowDoc_.setVisibilityState(visibilityState);
+    if (visibilityState != this.visibilityState_) {
+      this.shadowDoc_ && this.shadowDoc_.setVisibilityState(visibilityState);
       this.visibilityState_ = visibilityState;
```

Comment:
Feels a bit weird for the update of this property to be sandwiched between two side-effects. Would prefer to move this before the `setVisibilityState`.